### PR TITLE
Add Style/EmptyLineAfterMagicComment cop

### PR DESCRIPTION
**Pull request by backus**
Depends on bbatsov/ruby-style-guide#616

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

**Comment by backus** [comment]
Looks like this touches more tests than I thought. Work in progress....

File: spec/rubocop/cop/style/empty_line_after_magic_comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  it 'registers an offense for code that immediately follows comment' do
+    inspect_source(cop, ['# frozen_string_literal: true',
+                         'class Foo; end'])
+    expect(cop.offenses.size).to eq(1)
+    expect(cop.messages).to eq(['Add a space after magic comments.'])
+  end
+
+  it 'accepts code that separates the comment from the code with a newline' do
+    inspect_source(cop, ["# frozen_string_literal: true\n",
```
Comment (rrosenblum) [inline spec/rubocop/cop/style/empty_line_after_magic_comment_spec.rb]: What are your thoughts on writing this as

```ruby
inspect_source(cop, ["# frozen_string_literal: true",
                     '',
                     'class Foo; end'])
```

In my mind, it is a little clearer that there is an empty line between the comment and the class.

**Comment by rrosenblum** [comment]
Should this be configurable so that the user can choose whether they want to enforce having an empty line or not having an empty line?

I don't think that I saw a test that includes a magic comment and documentation comment. Should the following code register an offense?

```ruby
# frozen_string_literal: true
# Class documentation
class Foo
end
```

**Comment by Drenmi** [comment]
> I don't think that I saw a test that includes a magic comment and documentation comment. Should the following code register an offense?
> 
> ```
> # frozen_string_literal: true
> # Class documentation
> class Foo
> end
> ```

IMHO, it should. To me, the lack of spacing would indicate that both comment lines pertain to `Foo`, which is not the case, here.

File: spec/rubocop/cop/style/empty_line_after_magic_comment_spec.rb (hunk 1 above)
Comment (backus) [inline spec/rubocop/cop/style/empty_line_after_magic_comment_spec.rb]: Yeah I'm fine with either just didn't know what was the convention

**Comment by backus** [comment]
@bbatsov build is passing finally minus this weird jruby bug that is plaguing every build.

**Comment by backus** [comment]
@drenmi and @rrosenblum I think I addressed your concerns

**Pull request merged**